src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/utils/TestUtils.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.homeoffice.digital.sas.balancecalculator.utils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.TimeEntry;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+
+public class TestUtils {
+
+  public static KafkaEventMessage generateExpectedKafkaEventMessage(String version, TimeEntry resource,
+                                                                    KafkaAction action) {
+    return new KafkaEventMessage<>(version, resource, action);
+  }
+
+  public static TimeEntry createTimeEntry() {
```

Comment:
Tests are too fragile, we have one version of the time entry, when writing a test to ensure concurrent messages could be handled I set the actualStartTime dynamically and the test fell apart because it expected a hardcoded value.